docs/tutorials/using_datasets/IQL_torchrl.py

```diff
@@ -0,0 +1,522 @@
+
+# %%
+# For headless environments
+# ~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# %%
+# If you are in a headless environment (e.g. a Google Colab notebook), you will also need to install a virtual display. First, install the prerequisites:
+
+# %%
+# .. code-block:: bash
+#
+#     ! sudo apt-get update
+#     ! sudo apt-get install -y python3-opengl
+#     ! apt install ffmpeg
+#     ! apt install xvfb
+#     ! pip install pyvirtualdisplay
+#
+
+# %%
+# Then restart the notebook kernel. Once the pre-requisites are installed you can start a virtual display:
+
+# %%
+from pyvirtualdisplay import Display
+
+virtual_display = Display(visible=0, size=(1400, 900))
+virtual_display.start()
+
+# %%
+# The Adroit Pen environment
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# %%
+from torchrl.envs.libs.gym import GymEnv
+from torchrl.envs import DoubleToFloat, TransformedEnv
+
+# %%
+# We will be using the ``AdroitHandPen`` environment from `Gymnasium-Robotics <https://robotics.farama.org/envs/adroit_hand/adroit_pen/>`_. TorchRL is designed to be agnostic to different frameworks, so instead of working with a Gymnasium environment directly we load it using the ``GymEnv`` wrapper:
+
+# %%
+env_id = "AdroitHandPen-v1"
+example_env = GymEnv(env_id, from_pixels=True, pixels_only=False)
+
+# %%
+# ``GymEnv`` provides the usual methods such as ``env.step()`` and ``env.reset()``. However, instead of returning a tuple of step/reset data, they return a ``TensorDict``. A tensordict is essentially a dictionary of tensors whose first axis (the batch dimension) has the same size, and share some other properties like the device they are on. The tensordict returned has fields for each type of step data (e.g. ``observations``, ``actions``, ``rewards``):
+
+# %%
+tensordict = example_env.reset()
+print(tensordict)
+
+# %%
+# .. code-block:: text
+#
+#     TensorDict(
+#         fields={
+#             done: Tensor(shape=torch.Size([1]), device=cpu, dtype=torch.bool, is_shared=False),
+#             observation: Tensor(shape=torch.Size([45]), device=cpu, dtype=torch.float64, is_shared=False),
+#             pixels: Tensor(shape=torch.Size([480, 480, 3]), device=cpu, dtype=torch.uint8, is_shared=False),
+#             terminated: Tensor(shape=torch.Size([1]), device=cpu, dtype=torch.bool, is_shared=False),
+#             truncated: Tensor(shape=torch.Size([1]), device=cpu, dtype=torch.bool, is_shared=False)},
+#         batch_size=torch.Size([]),
+#         device=cpu,
+#         is_shared=False)
+
+# %%
+# \
+# TorchRL also provides ``env.rollout()`` to step through a full episode and save the step data in a tensordict:
+
+# %%
+max_episode_steps = 1000
+tensordict = example_env.rollout(max_steps=max_episode_steps, auto_cast_to_device=True)
+
+# %%
+# \
+# We can compute the cumulative reward by summing ``tensordict['next', 'reward']``:
+
+# %%
+print(f"Cumulative reward: {tensordict['next', 'reward'].sum():.2f}")
+
+# %%
+# .. code-block:: text
+#
+#     Cumulative reward: -31.10
+#
```

Comment:
I get a different value when running it (-871.17)

Reply:
It turns out I wasn't seeding everything. Thanks for spotting this!